trieste/models/keras/models.py

```diff
@@ -332,3 +334,37 @@ def optimize(self, dataset: Dataset) -> None:
 
         x, y = self.prepare_dataset(dataset)
         self.model.fit(x=x, y=y, **self.optimizer.fit_args)
+
+    def __getstate__(self) -> dict[str, Any]:
```

Comment:
a bit of education please :)
when are these getstate and setstate methods called?

Reply:
Yes. The getstate and setstate methods let you control pickling, and deepcopy defaults to pickling if there is no explicit deepcopy override. For a motivating example see https://docs.python.org/3/library/pickle.html#pickle-state